### DISCUSSION TITLE
How to expand ModelQuery results?

### DISCUSSION THREAD

**Discussion by sa7mon** [pocketbase/pocketbase#4791] [Q&A]
I've got 2 collections: `food` which contain many `products`. They look like this:

```shell
sqlite> PRAGMA table_info(food);
0|created|TEXT|1|strftime('%Y-%m-%d %H:%M:%fZ')|0
1|id|TEXT|1|'r'||lower(hex(randomblob(7)))|1
2|name|TEXT|1|''|0
3|products|JSON|1|'[]'|0
4|updated|TEXT|1|strftime('%Y-%m-%d %H:%M:%fZ')|0

sqlite> sqlite> PRAGMA table_info(products);
0|created|TEXT|1|strftime('%Y-%m-%d %H:%M:%fZ')|0
1|id|TEXT|1|'r'||lower(hex(randomblob(7)))|1
2|product_id|TEXT|1|''|0
3|store|TEXT|1|''|0
4|updated|TEXT|1|strftime('%Y-%m-%d %H:%M:%fZ')|0
5|price|NUMERIC|1|0|0
```

I'm looking to create the simplest function which returns all the `food` with their `products` expanded.

Using record collection methods, this is simple enough but yields `[]*models.Record`

```go
func GetAllFoodExpanded(dao *daos.Dao) ([]*models.Record, error) {
	records, err := dao.FindRecordsByFilter(
		"food",     // collection
		"id != ''", // filter
		"",         // sort
		1000,       // limit
		0,          // offset
		nil,
	)
	if err != nil {
		return nil, err
	}
	dao.ExpandRecords(records, []string{"products"}, nil)
	return records, nil
}
```

I can also use a ModelQuery to get a slice of `Food`, but as far as I know I'd need to make a ProductsQuery and manually merge the 2 slices.

```go
type Food struct {
	models.BaseModel

	Products []string `db:"expand.products" json:"products"`
	Name     string   `db:"name" json:"name"`
}

func GetAllFood(dao *daos.Dao) ([]*Food, error) {
	food := []*Food{}

	err := FoodQuery(dao).
		Limit(-1).
		OrderBy().
		All(&food)
	return food, err
}
```

Is there an easier way to do this that I'm missing?

**Comment by ganigeorgiev** [comment]
There is no easy way to do this at the moment, especially for slice of structs. You can find some information about the scan rules and an example with a child struct in the [dbx package readme](https://github.com/pocketbase/dbx?tab=readme-ov-file#executing-queries).

The question lacks details to understand the use case and why this is needed in the first place, so I'm not sure what to recommend but an easier alternative to populating eagerly the relation could be to have a helper getter method in the `Food` struct that will send query to fetch its `Products`. For most of the cases [N+1 queries are not a problem with SQLite](https://www.sqlite.org/np1queryprob.html).
